Assets/MRTK/Services/InputSystem/FocusProvider.cs

```diff
@@ -905,16 +909,39 @@ public bool UnregisterPointer(IMixedRealityPointer pointer)
         /// <inheritdoc />
         public IEnumerable<T> GetPointers<T>() where T : class, IMixedRealityPointer
         {
-            List<T> typePointers = new List<T>();
-            foreach (PointerData pointer in pointers.Values)
+            for (int i = 0; i < pointersList.Count; i++)
             {
-                if (pointer.Pointer is T typePointer && !typePointer.IsNull())
+                if (pointersList[i].Pointer is T typePointer && !typePointer.IsNull())
                 {
-                    typePointers.Add(typePointer);
+                    yield return typePointer;
```

Comment:
We discussed this change a while back in https://github.com/microsoft/MixedRealityToolkit-Unity/pull/8118. Any chance you did any perf measurements to verify that this isn't trading garbage for perf?

Reply:
Honestly, the best solution would be to have an overload where you can pass a list in, but that would be a breaking change. The comparison in the article is not exactly valid because here you were creating a List, iterating on it with foreach and returning the IEnumerable anyway.